### PR TITLE
TST: Fixed xfail for tests in pandas/tests/tseries/offsets/test_offsets_properties.py

### PR DESCRIPTION
**Pull request by ShaharNaveh**
- [x] ref https://github.com/pandas-dev/pandas/pull/32459#issuecomment-595277125
- [ ] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/tests/tseries/offsets/test_offsets_properties.py [hunk 1]
```diff
@@ -85,8 +85,6 @@
 # Offset-specific behaviour tests
 
 
-# Based on CI runs: Always passes on OSX, fails on Linux, sometimes on Windows
-@pytest.mark.xfail(strict=False, reason="inconsistent between OSs, Pythons")
```
Comment (MomIsBestFriend) [inline pandas/tests/tseries/offsets/test_offsets_properties.py]: It says here that on linux it always fails (at least this is how I understood it), but it passed on my machine.

File: pandas/tests/tseries/offsets/test_offsets_properties.py [hunk 2]
```diff
@@ -97,40 +95,39 @@ def test_on_offset_implementations(dt, offset):
...
+    res_v2 = offset.apply_index(rng.tz_localize(None)).tz_localize(rng.tz)
     assert (res == res_v2).all()
 
+    # apply_index is only for indexes, not series, so no res2_v2
     assert res[0] == rng[0] + offset
     assert res[-1] == rng[-1] + offset
     res2 = ser + offset
-    # apply_index is only for indexes, not series, so no res2_v2
+
+    # TODO: Check randomly assorted entries, not just first/last
```
Comment (MomIsBestFriend) [inline pandas/tests/tseries/offsets/test_offsets_properties.py]: I am interested in doing that, anyone got a smart way of doing that?

File: pandas/tests/tseries/offsets/test_offsets_properties.py [hunk 3]
```diff
@@ -97,40 +95,39 @@ def test_on_offset_implementations(dt, offset):
...
     assert res2.iloc[0] == ser.iloc[0] + offset
     assert res2.iloc[-1] == ser.iloc[-1] + offset
-    # TODO: Check randomly assorted entries, not just first/last
 
 
 @pytest.mark.xfail  # TODO: reason?
 @given(gen_yqm_offset)
 def test_shift_across_dst(offset):
```
Comment (MomIsBestFriend) [inline pandas/tests/tseries/offsets/test_offsets_properties.py]: I couldn't remove the ```xfail``` for this test case, @jbrockmendel mind taking a look?

Comment (jbrockmendel) [reply pandas/tests/tseries/offsets/test_offsets_properties.py]: @MomIsBestFriend I asked you to take a look, no backsies

File: pandas/tests/tseries/offsets/test_offsets_properties.py (hunk 1 above)
Comment (MomIsBestFriend) [reply pandas/tests/tseries/offsets/test_offsets_properties.py]: Ok so I think I understood why this failing:

```hypothesis.extra.dateutil.timezones``` is using ```dateutil.UTC``` which is avaible only from ```dateutils>=2.7.0``` (https://dateutil.readthedocs.io/en/stable/tz.html#dateutil.tz.dateutil.tz.UTC)

but macOS have a pinned ```python-dateutils``` to ```2.6.1``` which then leads to errors.

**Comment by MomIsBestFriend** [comment]
```mypy``` is failing?

File: pandas/tests/tseries/offsets/test_offsets_properties.py [hunk 4]
```diff
@@ -101,7 +103,7 @@ def test_on_offset_implementations(dt, offset):
 
 
 @pytest.mark.skipif(
-    dateutil.__version__ < "2.7",
+    DATEUTIL_VERSION < "2.7",
     reason="hypothesis uses dateutil.tz.UTC which was intreduced in dateutils 2.7.0",
```
Comment (jbrockmendel) [inline pandas/tests/tseries/offsets/test_offsets_properties.py]: intreduced -> introduced

File: pandas/tests/tseries/offsets/test_offsets_properties.py [hunk 5]
```diff
@@ -101,7 +103,7 @@ def test_on_offset_implementations(dt, offset):
 
 
 @pytest.mark.skipif(
-    dateutil.__version__ < "2.7",
+    DATEUTIL_VERSION < "2.7",
```
Comment (jbrockmendel) [inline pandas/tests/tseries/offsets/test_offsets_properties.py]: can you use LooseVersion

File: pandas/tests/tseries/offsets/test_offsets_properties.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/tests/tseries/offsets/test_offsets_properties.py]: > is bumping python-dateutils to 2.7.0 is an option/reasonable thing to do?

2.7.0 is 2 days shy of 2 years old, so I think we're OK bumping

**Comment by jbrockmendel** [comment]
looks like the build logs have expired; can you rebase and we'll see if this is ready

File: pandas/tests/tseries/offsets/test_offsets_properties.py [hunk 6]
```diff
@@ -85,8 +89,10 @@
 # Offset-specific behaviour tests
 
 
-# Based on CI runs: Always passes on OSX, fails on Linux, sometimes on Windows
-@pytest.mark.xfail(strict=False, reason="inconsistent between OSs, Pythons")
+@pytest.mark.skipif(
```
Comment (simonjayhawkins) [inline pandas/tests/tseries/offsets/test_offsets_properties.py]: use td.skip_if_no with min_version

Comment (MomIsBestFriend) [reply pandas/tests/tseries/offsets/test_offsets_properties.py]: Thank you, didn't even know that decorator exist.

File: pandas/tests/tseries/offsets/test_offsets_properties.py [hunk 7]
```diff
@@ -97,40 +98,43 @@ def test_on_offset_implementations(dt, offset):
...
+
+    # TODO: test for that case separately
+    assume(offset.n != 0)
+
     ser = pd.Series(rng)
 
     res = rng + offset
     res_v2 = offset.apply_index(rng)
-    # res_v2 = offset.apply_index(rng.tz_localize(None)).tz_localize(rng.tz)
+    res_v2 = offset.apply_index(rng.tz_localize(None)).tz_localize(rng.tz)
```
Comment (jbrockmendel) [inline pandas/tests/tseries/offsets/test_offsets_properties.py]: delete the line above?

File: pandas/tests/tseries/offsets/test_offsets_properties.py [hunk 8]
```diff
@@ -97,40 +98,43 @@ def test_on_offset_implementations(dt, offset):
...
-    "has errors in timezone generation."
-)
+@td.skip_if_no("dateutil", "2.7")
 @given(gen_yqm_offset, gen_date_range)
 def test_apply_index_implementations(offset, rng):
     # offset.apply_index(dti)[i] should match dti[i] + offset
-    assume(offset.n != 0)  # TODO: test for that case separately
 
-    # rng = pd.date_range(start='1/1/2000', periods=100000, freq='T')
+    # TODO: Remove this `warnings.simplefilter`
```
Comment (jbrockmendel) [inline pandas/tests/tseries/offsets/test_offsets_properties.py]: what are the conditions that will allow this?

File: pandas/tests/tseries/offsets/test_offsets_properties.py (hunk 2 above)
Comment (jbrockmendel) [reply pandas/tests/tseries/offsets/test_offsets_properties.py]:
```
for _ in range(10):
     n = np.random.randint(len(ser))
     assert res2.iloc[n] == ser.iloc[n] + offset
```

(10 is arbitrary number here, you decide what is reasonable number

**Comment by jbrockmendel** [comment]
Couple of comments, looks nearly there

File: pandas/tests/tseries/offsets/test_offsets_properties.py (hunk 8 above)
Comment (MomIsBestFriend) [reply pandas/tests/tseries/offsets/test_offsets_properties.py]: I can't remebmer why was this necessery, Ill remove it, and see if the CI passes

File: pandas/tests/tseries/offsets/test_offsets_properties.py (hunk 3 above)
Comment (jbrockmendel) [reply pandas/tests/tseries/offsets/test_offsets_properties.py:127]: so no luck here?

File: pandas/tests/tseries/offsets/test_offsets_properties.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/tests/tseries/offsets/test_offsets_properties.py:89]: @MomIsBestFriend this is definitely an improvement, but we can also just bump the dateutil requirement

Comment (MomIsBestFriend) [reply pandas/tests/tseries/offsets/test_offsets_properties.py:89]: > @MomIsBestFriend this is definitely an improvement, but we can also just bump the dateutil requirement

Sure! will do that in a separate PR

File: pandas/tests/tseries/offsets/test_offsets_properties.py (hunk 3 above)
Comment (MomIsBestFriend) [reply pandas/tests/tseries/offsets/test_offsets_properties.py:127]: > so no luck here?

No :disappointed: I could not make this stable/work properly

**Comment by jbrockmendel** [comment]
```
=================================== FAILURES ===================================
_______________________ test_apply_index_implementations _______________________
[gw1] linux -- Python 3.6.10 /home/vsts/miniconda3/envs/pandas-dev/bin/python

    @given(gen_yqm_offset, gen_date_range)
>   def test_apply_index_implementations(offset, rng):

pandas/tests/tseries/offsets/test_offsets_properties.py:99: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
pandas/core/indexes/datetimes.py:974: in date_range
    **kwargs,
pandas/core/arrays/datetimes.py:412: in _generate_range
    index.asi8, tz, ambiguous=ambiguous, nonexistent=nonexistent
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

>   raise pytz.NonExistentTimeError(stamp)
E   pytz.exceptions.NonExistentTimeError: 1991-03-31 00:04:11.080387

pandas/_libs/tslibs/tzconversion.pyx:276: NonExistentTimeError
```

File: pandas/tests/tseries/offsets/test_offsets_properties.py (hunk 3 above)
Comment (jbrockmendel) [reply pandas/tests/tseries/offsets/test_offsets_properties.py:123]: I appreciate you taking a look.  With enough eyeballs we'll eventually get this figured out.

**Comment by alimcmaster1** [comment]
Is this still active?

**Comment by jreback** [comment]
closing as stale; if you can continue pls ping / open a new PR.

**Pull request closed without merging**